schema_based_on_diagram.sql

```diff
@@ -0,0 +1,42 @@
+CREATE TABLE "medical_histories"(
+"id" SERIAL PRIMARY KEY,
+"admitted_at" TIMESTAMP WITHOUT TIME ZONE NOT NULL,
+"patient_id" INTEGER NOT NULL REFERENCES "patients"("id"),
+"status" VARCHAR(255) NOT NULL,
+INDEX("patient_id"),
+INDEX("id")
+);
+
+CREATE TABLE "invoices"(
+"id" SERIAL PRIMARY KEY,
+"total_amount" DECIMAL(8, 2) NOT NULL,
+"generated_at" TIMESTAMP WITHOUT TIME ZONE NOT NULL,
+"paid_at" TIMESTAMP WITHOUT TIME ZONE,
+"medical_history_id" INTEGER NOT NULL REFERENCES "medical_histories"("id"),
+INDEX("medical_history_id")
+);
+
+CREATE TABLE "patients"(
+"id" SERIAL PRIMARY KEY,
+"name" VARCHAR(255) NOT NULL,
+"date_of_birth" DATE NOT NULL
+);
+
+CREATE TABLE "treatments"(
+"id" SERIAL PRIMARY KEY,
+"type" VARCHAR(255) NOT NULL,
+"name" VARCHAR(255) NOT NULL,
+"medical_history_id" INTEGER NOT NULL REFERENCES "medical_histories"("id"),
+INDEX("medical_history_id")
+);
```

Comment:
- [ ] There many too many relationships between `treatments` and `medical_histrories` as shown in  the diagram and in order to achieve that you need to add a table between them that will have the foreign keys  please make sure  they have indexes  as required